src/main/java/org/sunbird/workflow/service/impl/UserProfileWfServiceImpl.java

```diff
@@ -432,6 +432,16 @@ private void failedCase(WfRequest wfRequest) {
 	private Map<String, Object> getUpdateRequest(WfRequest wfRequest, Map<String, Object> updateRequest) {
 		Map<String, Object> requestObject = new HashMap<>();
 		Map<String, Object> requestWrapper = new HashMap<>();
+		List<HashMap<String, Object>> valuesToBeUpdate = wfRequest.getUpdateFieldValues();
+		for (Map<String, Object> valueToUpdate : valuesToBeUpdate) {
+			if(valueToUpdate.containsKey(Constants.TO_VALUE)) {
+				Map<String, Object> toValue = (Map<String, Object>) valueToUpdate.get(Constants.TO_VALUE);
+				String keyToUpdate = toValue.keySet().stream().findFirst().get();
+				if(Constants.FIRSTNAME.equalsIgnoreCase(keyToUpdate)) {
+					requestWrapper.put(Constants.FIRST_NAME_CAMEL_CASE, toValue.get(keyToUpdate));
```

Comment:
currently using  "profileDetails.personalDetails.firstname" -- not "firstName"  -- as per the ES user mapping... 
let's use firstname only -- though the incoming value is firstName - change it to firstname

also, the personal details needs to be updated for other fields too... 
mobile / primaryEmail -- those fields also you suppose to check right ??

Reply:
Both firstname(in personal details) and firstName(root level) is used Karthik.